parsing/ast_helper.ml

```diff
@@ -114,6 +115,8 @@ module Typ = struct
             Ptyp_poly(string_lst, loop core_type)
         | Ptyp_package(longident,lst) ->
             Ptyp_package(longident,List.map (fun (n,typ) -> (n,loop typ) ) lst)
+        | Ptyp_open (mod_ident, core_type) ->
+            Ptyp_open (mod_ident, core_type)
```

Comment:
This is missing a `loop core_type`. Otherwise
```ocaml
let f: type a. Int.( a -> a ) = fun x -> x;;
```
fails to transform `a` into `'a` yielding the following error:

> Error: Unbound type constructor a